examples/web-server/caddy/README.md

```diff
@@ -0,0 +1,127 @@
+   - `*.coder.example.com`: Domain name for wildcard apps, commonly used for [dashboard port forwarding](https://coder.com/docs/coder-oss/latest/networking/port-forwarding#dashboard). This is optional and can be removed.
+
+1. Start Coder. Set `CODER_ACCESS_URL` and `CODER_WILDCARD_ACCESS_URL` to the domain you're using in your Caddyfile.
+
+   ```sh
+   export CODER_ACCESS_URL=https://coder.example.com
+   export CODER_WILDCARD_ACCESS_URL=*.coder.example.com
+   docker compose up -d # Run on startup
+   ```
+
+### Standalone
+
+1. If you haven't already, [install Coder](https://coder.com/docs/coder-oss/latest/install)
+
+1. Install [Caddy Server](https://caddyserver.com/docs/install)
+
+1. Copy our sample [Caddyfile](./Caddyfile) and change the following values:
+
+   > If you're installed Caddy as a system package, update the default Caddyfile with `vim /etc/caddy/Caddyfile`
+
+   - `email@example.com`: Email to request certificates from LetsEncrypt/ZeroSSL (does not have to be Coder admin email)
+   - `coder.example.com`: Domain name you're using for Coder.
+   - `*.coder.example.com`: Domain name for wildcard apps, commonly used for [dashboard port forwarding](https://coder.com/docs/coder-oss/latest/networking/port-forwarding#dashboard). This is optional and can be removed.
+   - `localhost:3000`: Address Coder is running on. Modify this if you changed `CODER_ADDRESS` in the Coder configuration.
+
+1. [Configure Coder](https://coder.com/docs/coder-oss/latest/admin/configure) and change the following values:
+
+   - `CODER_ACCESS_URL`: root domain (e.g. `https://coder.example.com`)
+   - `CODER_WILDCARD_ACCESS_URL`: wildcard domain (e.g. `*.example.com`).
+
+1. Start the Caddy server:
+
+   If you're [keeping Caddy running](https://caddyserver.com/docs/running) via a system service:
+
+   ```sh
+   sudo systemctl restart caddy
+   ```
+
+   Or run a standalone server:
+
+   ```sh
+   caddy run
+   ```
+
+1. Optionally, use [ufw](https://wiki.ubuntu.com/UncomplicatedFirewall) or another firewall to disable external traffic outside of Caddy.
+
+   ```sh
+   # Check status of UncomplicatedFirewall
+   sudo ufw status
+
+   # Allow SSH
+   sudo ufw allow 22
+
+   # Allow HTTP, HTTPS (Caddy)
+   sudo ufw allow 80
+   sudo ufw allow 443
+
+   # Deny direct access to Coder server
+   sudo ufw deny 3000
+
+   # Enable UncomplicatedFirewall
+   sudo ufw enable
+   ```
+
+1. Navigate to your Coder URL! A TLS certificate should be auto-generated on your first visit.
+
+## Generating wildcard certificates
+
+By default, this configuration uses Caddy's [on-demand TLS](https://caddyserver.com/docs/caddyfile/options#on-demand-tls) to generate a certificate for each subdomain (e.g. `app1.coder.example.com`, `app2.coder.example.com`). When users visit new subdomains, such as accessing [ports on a workspace](../../networking/port-forwarding.md), the request will take an additional 5-30 seconds since a new certificate is being generated.
```

Comment:
> the request will take an additional 5-30 seconds

I'm fairly certain this is the only downside of using Caddy's on-demand TLS. While LetsEncrypt's rate limit (50 certificates/domain/week) can quickly be exhausted, Caddy will silently fall back to ZeroSSL which has no rate limit. ZeroSSL is significantly slower though. If you think this needs further explanation in the docs, I can expand.

With that being said, the extra effort for a wildcard is worth it if a Coder deployment is being actively being used in production.